test/new_relic/agent/log_event_aggregator_test.rb

```diff
@@ -521,5 +527,164 @@ def test_records_log_events_not_within_default_severities
+
+        assert_empty events
+      end
+    end
+
+    def test_records_customer_metrics_when_enabled_logstasher
+      with_config(LogEventAggregator::METRICS_ENABLED_KEY => true) do
+        2.times { @aggregator.record_logstasher_event({'message' => 'metrics enabled', 'level' => :debug}) }
+        @aggregator.harvest!
+      end
+
+      assert_metrics_recorded({
+        'Logging/lines' => {:call_count => 2},
+        'Logging/lines/debug' => {:call_count => 2}
+      })
+    end
+
+    def test_doesnt_record_customer_metrics_when_overall_disabled_and_metrics_enabled_logstasher
+      with_config(
+        LogEventAggregator::OVERALL_ENABLED_KEY => false,
+        LogEventAggregator::METRICS_ENABLED_KEY => true
+      ) do
+        NewRelic::Agent.config.notify_server_source_added
+
+        @aggregator.record_logstasher_event({'message' => 'overall disabled, metrics enabled', 'level' => :debug})
+        @aggregator.harvest!
+      end
+
+      assert_metrics_not_recorded([
+        'Logging/lines',
+        'Logging/lines/debug'
+      ])
+    end
+
+    def test_doesnt_record_customer_metrics_when_disabled_logstasher
+      with_config(LogEventAggregator::METRICS_ENABLED_KEY => false) do
+        @aggregator.record_logstasher_event({'message' => 'metrics disabled', 'level' => :warn})
+        @aggregator.harvest!
+      end
+
+      assert_metrics_not_recorded([
+        'Logging/lines',
+        'Logging/lines/warn'
+      ])
+    end
+
+    def test_high_security_mode_logstasher
+      with_config(CAPACITY_KEY => 5, :high_security => true) do
+        # Refresh the high security setting on this notification
+        NewRelic::Agent.config.notify_server_source_added
+
+        @aggregator.record_logstasher_event({'message' => 'high security enabled', 'level' => :info})
+        _, events = @aggregator.harvest!
+
+        assert_empty events
+      end
+    end
+
+    def test_does_not_record_logstasher_events_with_a_severity_below_config
+      with_config(LogEventAggregator::LOG_LEVEL_KEY => 'info') do
+        assert_equal :INFO, @aggregator.send(:configured_log_level_constant)
+
+        @aggregator.record_logstasher_event({'message' => 'severity below config', 'level' => :debug})
+        _, events = @aggregator.harvest!
+
+        assert_empty events
+      end
+    end
+
+    def test_record_logstasher_exits_if_forwarding_disabled
+      with_config(LogEventAggregator::FORWARDING_ENABLED_KEY => false) do
+        @aggregator.record_logstasher_event({'message' => 'forwarding disabled', 'level' => :info})
+        _, results = @aggregator.harvest!
+
+        assert_empty results
+      end
+    end
+
+    def test_nil_when_record_logstasher_errors
+      @aggregator.stub(:severity_too_low?, -> { raise 'kaboom' }) do
+        @aggregator.record_logstasher_event({'level' => :warn, 'message' => 'A trex is near'})
+        _, results = @aggregator.harvest!
+
+        assert_empty results
+      end
+    end
```

Comment:
This gets me curious about what the logstasher log events look like when an error is raised. The spec has some special attributes to send in those scenarios (ex. `error.stack`), but I don't believe we use them in the logger instrumentation.

IIRC there were some special attributes when logstasher had an error. How does logstasher behave if the log has a related error?

Reply:
If LogStasher has an error, it adds an `error` key to the hash with the stack trace as the value. I wasn't able to find anything in Logger instrumentation that suggested we do anything special with errors. 

I'm torn between molding the LogStasher data to fit our spec, or simply reporting their data as it is. For example with errors, right now we currently report `error` like any other attribute and it includes the info a LogStasher user might expect. To fit the spec, I'd break the error into two new key/values: `error.stack` and `error.class`.

Reply:
We chatted a little more about this one synchronously. I like leaning on reporting the data as logstasher does, but am open to the approach that's closer to the spec as well.

Reply:
Chatted live - lets keep closer to how logstasher reports data.